src/ResourceManagement/BatchAI/BatchAIJobImpl.cs

```diff
@@ -290,6 +441,31 @@ public CaffeSettings CaffeSettings()
             return Inner.CaffeSettings;
         }
 
+        ///GENMHASH:446B156BB626194DD2A3F46642818AF8:551F630EC4E164B1BDC4E1F286DA3951
+        public BatchAIJobImpl WithEnvironmentVariable(string name, string value)
+        {
+            EnsureEnvironmentVariables().Add(new EnvironmentVariable(name, value));
+            return this;
+        }
+
+        ///GENMHASH:997F16B1AEBB01217D1CAE2B03B8B73E:E4346C9A19D9DCFF2986204E3D4749B1
+        public BatchAIJobImpl WithEnvironmentVariableSecretValue(string name, string value)
+        {
+            EnsureEnvironmentVariablesWithSecrets().Add(new EnvironmentVariableWithSecretValue(name, value));
+            return this;
+        }
+
+                ///GENMHASH:AB0BF9D0BEA18CD334AAF69A466D74DB:244EAB58A02E8C0EA8858C0AA028A77B
+        public BatchAIJobImpl WithEnvironmentVariableSecretValue(string name, string keyVaultId, string secretUrl)
+        {
+                    //$ KeyVaultSecretReference secretReference = new KeyVaultSecretReference()
+                    //$ .WithSourceVault(new ResourceId().WithId(keyVaultId)).WithSecretUrl(secretUrl);
+                    //$ ensureEnvironmentVariablesWithSecrets().Add(new EnvironmentVariableWithSecretValue().WithName(name).WithValueSecretReference(secretReference));
+                    //$ return this;
```

Comment:
Why is this code commented? Please either implement or add a text comment with the description why it is not implemented yet.